brushfire-features/src/main/scala/com/stripe/brushfire/features/FeatureParser.scala

```diff
@@ -0,0 +1,59 @@
+package com.stripe.brushfire
+package features
+
+trait FeatureParser[-A] {
+  def parse(value: A): Map[String, FValue]
+}
+
+trait TrainingDataParser[-A, T] {
+  def featureParser: FeatureParser[A]
+  def parseTimestamp(a: A): Long
+  def parseId(a: A): String
+  def parseTarget(a: A): T
+}
+
+case class CsvRowFeatureParser(header: CsvRow) extends FeatureParser[CsvRow] {
+  private val headerWithIndex: Vector[(String, Int)] = header.zipWithIndex.toVector
+
+  def parse(row: CsvRow): Map[String, FValue] =
+    headerWithIndex.map { case (feature, i) =>
+      feature -> FValue.dynamic(row(i))
+    } (collection.breakOut)
+}
+
+object CsvRowFeatureParser {
+  def trainer(platform: TrainingPlatform)(header: CsvRow): platform.Trainer[CsvRow, FeatureParser[CsvRow]] =
+    platform.Trainer.pure(CsvRowFeatureParser(header))
+}
+
+case class CsvRowTrainingDataParser(
+  header: CsvRow,
+  timestamp: Option[String],
+  id: Seq[String],
+  target: String
+) extends TrainingDataParser[CsvRow, Map[String, Long]] {
+  private def indexOf(tpe: String)(col: String): Int = {
```

Comment:
small thing but I found the currying here hard to read; maybe a slightly more verbose use of a single argument list would be better